moto/dynamodb2/models.py

```diff
@@ -110,27 +111,27 @@ def describe_attrs(self, attributes):
         }
 
     def update(self, update_expression, expression_attribute_names, expression_attribute_values):
-        ACTION_VALUES = ['SET', 'set', 'REMOVE', 'remove']
-
-        action = None
-        for value in update_expression.split():
-            if value in ACTION_VALUES:
-                # An action
-                action = value
-                continue
-            else:
+        parts = [p for p in re.split(r'\b(SET|REMOVE|ADD|DELETE)\b', update_expression) if p]
```

Comment:
I think this would be the `findall` version:
```
        operator_expr = r'\b(?:SET|REMOVE|ADD|DELETE)\b'
        parts = re.findall('({})((?:.(?!{}))+)'.format(operator_expr, operator_expr), update_expression)
```